code/include/playfab/PlayFabEventsDataModels_c.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <playfab/PlayFabBaseModel_c.h>
+
+#pragma push_macro("IN")
+#undef IN
+
+// Events enums
+
+// Events structs
+typedef struct PlayFabEventsEntityKey
+{
+    const char* id;
+    const char* type;
+
+} PlayFabEventsEntityKey;
+
+typedef struct PlayFabEventsEventContents
+{
+    PF_MAP struct PlayFabStringDictionaryEntry* customTags;
+    PF_COLLECTION_COUNT uint32_t customTagsCount;
+    PF_OPTIONAL PlayFabEventsEntityKey* entity;
+    const char* eventNamespace;
+    const char* name;
+    const char* originalId;
+    PF_OPTIONAL time_t* originalTimestamp;
+    PlayFabJsonObject payload;
+    const char* payloadJSON;
```

Comment:
does it need both a JSON object of payload and a char* payloadJSON?

Reply:
will look at service spec to see what these are.

---
In reply to: [604529115](https://github.com/jplafonta/XPlatCSdk/pull/1#discussion_r604529115) [](ancestors = 604529115)

Reply:
From the service contract:

"name": "Payload",
"description": "Arbitrary data associated with the event. Only one of Payload or PayloadJSON is allowed."

"name": "PayloadJSON",
"description": "Arbitrary data associated with the event, represented as a JSON serialized string. Only one of Payload or PayloadJSON is allowed."

Since we are representing both as strings, these are redundant.  Not technically harmful to include both in the struct if its documented, but it could definitely be confusing.  I will remove one

---
In reply to: [604539497](https://github.com/jplafonta/XPlatCSdk/pull/1#discussion_r604539497) [](ancestors = 604539497,604529115)